sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -809,6 +846,22 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     }
   }
 
+  override def getDatabaseOwnerName(db: Database): String = {
+    Option(getDatabaseOwnerNameMethod.invoke(db)).map(_.asInstanceOf[String]).getOrElse("")
+  }
+
+  override def setDatabaseOwnerName(db: Database, owner: String): Unit = {
+    setDatabaseOwnerNameMethod.invoke(db, owner)
+  }
+
+  override def getDatabaseOwnerType(db: Database): String = {
+    Option(getDatabaseOwnerTypeMethod.invoke(db))
+      .map(_.asInstanceOf[PrincipalType].name()).getOrElse(PrincipalType.USER.name())
```

Comment:
IIUC this is a corner case that when hive table doesn't have owner. I think it's more consistent to return "" if owner is null. We don't care too much about the behavior of a very old hive version.

Reply:
agree